superset-frontend/src/components/ListView/ListViewStyles.less

```diff
@@ -134,6 +134,10 @@
     }
 
     .row-count-container {
+      float: right;
```

Comment:
let's just change this to 
```css
float: right;
margin-right: 20px;
```

which should be enough to make this look OK on both the `/superset/welcome` and the `/tablemodelview/list/` pages with `ENABLE_REACT_CRUD_VIEWS` flag enabled. The `LIST_VIEWS_NEW_UI` flag is only related to the filters ui, you can update it to `LIST_VIEWS_NEW_FILTER_UI` to be more semantically correct if you're up for it.